Add Comments and Discussion Feature for Code Snippets #18

Vivek7038 opened the pull request:
- users can add and see comments for any snippets 
- for now comment author img is dummy 
-

notsoocool commented:
A few things I noticed @Vivek7038 :

1. **Error when posting a comment**: As soon as I click "Post" for a comment, it gives an error, but after refreshing, the comment is successfully shown. Can you check this behavior?
2. **Comments placement**: Comments and posting a comment should not be included inside the snippet card. They should be displayed **below** the cards.
3. **Dynamic updates**: The comments should dynamically update without the need for reloading the page.
4. **API structure**: It looks like there’s an extra API folder `//snippets/comments/route.ts`. Since you're using `//snippets/[id]/comments/route.ts` everywhere, please ensure you're not duplicating routes.

Please address these points and update the PR when you get a chance. Thanks!

Vivek7038 commented:
@notsoocool 

1  for me its working without page refresh 
2 i have added comments in seperate card 
3. i am adding the latest added comment in the ui directly without the get call again
4. deleted duplicate folder of comments 

![image](https://github.com/user-attachments/assets/34c4d810-d7be-4b76-b3da-d6dbd50c72d2)

Vivek7038 commented:
@notsoocool  i have pushed changes , please review it

notsoocool commented:
@Vivek7038 
It's still showing the same error with the same refresh update. Could you do one thing—record your screen and show me how it's working on your device? It shouldn't behave differently on your device compared to mine.

Vivek7038 commented:
> @Vivek7038 It's still showing the same error with the same refresh update. Could you do one thing—record your screen and show me how it's working on your device? It shouldn't behave differently on your device compared to mine. <img alt="Screenshot 2024-10-07 at 12 21 21 PM" width="930" src="https://private-user-images.githubusercontent.com/35025375/374063358-39344b03-1bb6-4f41-b224-698481bf1097.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgyODQ3NzQsIm5iZiI6MTcyODI4NDQ3NCwicGF0aCI6Ii8zNTAyNTM3NS8zNzQwNjMzNTgtMzkzNDRiMDMtMWJiNi00ZjQxLWIyMjQtNjk4NDgxYmYxMDk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA3VDA3MDExNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2ZjBiNDkyZDU4MDY1YmViZTQxNzk0ZTNkZDdkYzE0MTE3OWU1MTY4NzE0NTQxNGQwY2U5OTkwZWMwNWVmYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9zUE4LZHdlWgYJJbh3YnJbsZ8pGE2IgRL4rapQuKTS8">

https://www.loom.com/share/3f3681f23d724d72a314101d906516f2?sid=b50f385c-bef3-47c9-bd88-955d224d4a23

notsoocool commented on lib/db/snippetModel.ts:
It is giving error because you have added a comments field in snippet model. You have already created a different model and added a ref to snippet model, then why adding it here.

notsoocool commented on app/api/snippets/[id]/comments/route.ts:
```diff
@@ -0,0 +1,74 @@
+
+    snippet.comments = snippet.comments || [];
+    snippet.comments.push(newComment._id as Types.ObjectId);
+    await snippet.save();
```
and when you are saving the comment using the snippet model it is asking for the required field i.e category and all, which is giving the error.

notsoocool commented:
Please remove the comments field from the snippet model and update the code accordingly and the issue will be solved, and before that please fetch the recent changes from the develop field

notsoocool commented:
@Vivek7038 any update?

Vivek7038 commented:
> @Vivek7038 any update?

will commit latest change wait

notsoocool commented:
@Vivek7038 Perfect, it's working now! I'll do a final review of the PR this evening, and then I’ll merge it. Thank you for your contribution!
